File: src/main/java/com/example/entity/Burger.java
package com.example.entity;

public class Burger {
    private int id;
    private String nom;
    private double prix;
    private String url_image;
    private String description;
    private EtatStockEnum etatStock;

    public Burger(String nom, double prix, String url_image, String description) {
        this.nom = nom;
        this.prix = prix;
        this.url_image = url_image;
        this.description = description;
        this.etatStock = EtatStockEnum.disponible;
    }

    public Burger(int id, String nom, double prix, String url_image, String description, EtatStockEnum etatStock) {
        this.id = id;
        this.nom = nom;
        this.prix = prix;
        this.url_image = url_image;
        this.description = description;
        this.etatStock = etatStock;
    }
    


    public int getId() {
        return id;
    }

    public String getNom() {
        return nom;
    }

    public double getPrix() {
        return prix;
    }

    public String getUrl_image() {
        return url_image;
    }

    public String getDescription() {
        return description;
    }

    public EtatStockEnum getEtatStock() {
        return etatStock;
    }

 
    public void setId(int id) {
        this.id = id;
    }

    public void setNom(String nom) {
        this.nom = nom;
    }

    public void setPrix(double prix) {
        this.prix = prix;
    }

    public void setUrl_image(String url_image) {
        this.url_image = url_image;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public void setEtatStock(EtatStockEnum etatStock) {
        this.etatStock = etatStock;
    }


    @Override
    public String toString() {
        return "Burger{" +
                "id=" + id +
                ", nom='" + nom + '\'' +
                ", prix=" + prix +
                ", url_image='" + url_image + '\'' +
                ", description='" + description + '\'' +
                ", etatStock=" + etatStock +
                '}';
    }
}

File: Entity/Burger.cs
namespace BrasilBurger.Web.Entity
{
    public class Burger
    {
        public int Id { get; set; }
        public string Nom { get; set; }
        public double Prix { get; set; }
        public string? UrlImage { get; set; }
        public string? Description { get; set; }
        public EtatStockEnum EtatStock { get; set; }

        public Burger()
        {
            EtatStock = EtatStockEnum.disponible;
        }

        public Burger(string nom, double prix, string? urlImage, string? description)
        {
            Nom = nom;
            Prix = prix;
            UrlImage = urlImage;
            Description = description;
            EtatStock = EtatStockEnum.disponible;
        }

        public override string ToString()
        {
            return $"Burger{{Id={Id}, Nom='{Nom}', Prix={Prix}, EtatStock={EtatStock}}}";
        }
    }
}
